src/DynamoCoreWpf/Views/Core/NodeView.xaml

```diff
@@ -535,18 +549,90 @@
                 IsHitTestVisible="False"
                 SnapsToDevicePixels="True" />
 
-        <!--  Displays when the node is Frozen/Warning/Error state  -->
-        <Border Name="nodeColorOverlay"
+        <!--  Displays when the node is Frozen/Warning/Error state and Zoom is above 0.4 -->
+        <Border Name="nodeColorOverlayZoomIn"
                 Grid.Row="1"
                 Grid.RowSpan="4"
                 Grid.ColumnSpan="3"
                 Margin="-8"
                 Background="{StaticResource NodeFrozenOverlayColor}"
-                Canvas.ZIndex="7"
+                Canvas.ZIndex="6"
+                CornerRadius="8,8,0,0"
+                IsHitTestVisible="False"
+                Opacity="{Binding Path=DataContext.Zoom, 
```

Comment:
can you verify that in addition to lowering the opacity if the zoom level is high, that you are also making the item collapsed? This should remove the elements from the visual tree, avoiding layout and render costs.

Reply:
Sure thing, will do!

Reply:
I have introduced the changes, let me know if that hasn't translated into the desired increase in performance.